src/rust/ide/src/ide/integration.rs

```diff
@@ -1045,18 +1047,23 @@ impl Model {
         let module_name          = crate::ide::INITIAL_MODULE_NAME;
         let visualisation_module = QualifiedName::from_segments(project_name,&[module_name])?;
         let id                   = VisualizationId::new_v4();
-        let expression           = crate::constants::SERIALIZE_TO_JSON_EXPRESSION.into();
+        let metadata_expression  = metadata.map(|m| m.expression.as_ref().map(|e| e.to_string()));
+        let metadata_expression  = metadata_expression.flatten();
```

Comment:
You may use https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then